safe-svg.php

```diff
@@ -270,11 +270,13 @@ public function fix_admin_preview( $response, $attachment, $meta ) {
          */
         public function one_pixel_fix( $image, $attachment_id, $size, $icon ) {
             if ( get_post_mime_type( $attachment_id ) === 'image/svg+xml' ) {
-                if ( empty( $image[1] ) ) {
-                    $image[1] = 100;
-                }
+                $dimensions = $this->svg_dimensions( get_attached_file( $attachment_id ) );
 
-                if ( empty( $image[2] ) ) {
+                if ( $dimensions ) {
+                    $image[1] = $dimensions['width'];
```

Comment:
Shouldn't we calculate the image width and height respecting the `$size` here?

For example, when calling `wp_get_attachment_image_src` filter for the `'medium'` size image we expect to reduce or increase width and height to the max 300px side by default (like it works for regular JPG/PNG).

With the current change, having a SVG with original dimensions of 1200x500 will produce `<img src="/image.svg" width="1200" height="500" />` regardless the requested `$size`

Reply:
> With the current change, having a SVG with original dimensions of 1200x500 will produce <img src="/image.svg" width="1200" height="500" /> regardless the requested $size

As far as I can tell in my testing, this is how it's been working prior to the fix we released in 1.9.10, so this PR attempts to bring that same functionality back.

I do agree that it seems like we should be taking into account the actual requested image size and use that to determine the size of the SVG but I believe that will break backwards compatibility, though maybe we're fine with that? The other issue to think about is SVGs don't get resized like normal images, so there aren't cropped versions of SVGs to match each registered image size like happens with normal image types.

If someone requests a medium image size, we can't just return 300x300 as that could end up returning a much larger SVG or could end up making the SVG stretched in one direction or the other. There would need to be some more logic added here that takes the requested image size and tries to find the closest dimensions that match the original SVG height/width ratio (and ensure we don't return dimensions that are greater than the actual SVG dimensions).

I think this would be good to fix but I'm leaning towards having that in a separate PR, as the approach in this PR is to bring us back to where we were prior to v1.9.10 but still fix the original issue that we were attempting to fix